core/src/test/java/nl/nn/adapterframework/jta/narayana/NarayanaJtaTransactionManagerTest.java

```diff
@@ -0,0 +1,206 @@
+		if (delegateTransactionManager != null) {
+			delegateTransactionManager.shutdownTransactionManager();
+			delegateTransactionManager = null;
+		}
+		log.debug("end teardown");
+	}
+
+	private NarayanaJtaTransactionManager getNarayanaJtaTransactionManager() throws Exception {
+
+		log.debug("getNarayanaJtaTransactionManager folder ["+folder.getRoot().toString()+"]");
+
+		NarayanaJtaTransactionManager result = new NarayanaJtaTransactionManager();
+		Properties props = new Properties();
+		props.setProperty("JDBCEnvironmentBean.isolationLevel", "2" );
+		props.setProperty("ObjectStoreEnvironmentBean.objectStoreDir", folder.getRoot().toString());
+		props.setProperty("ObjectStoreEnvironmentBean.stateStore.objectStoreDir", folder.getRoot().toString());
+		props.setProperty("ObjectStoreEnvironmentBean.communicationStore.objectStoreDir", folder.getRoot().toString());
+
+		NarayanaConfigurationBean config = new NarayanaConfigurationBean();
+		config.setProperties(props);
+		config.afterPropertiesSet();
+
+		result.setStatusFile(folder.getRoot().toString()+"/"+STATUS_FILE);
+		result.setUidFile(folder.getRoot().toString()+"/"+TMUID_FILE);
+		delegateTransactionManager = result;
+		return result;
+	}
+
+	@Test
+	public void testCleanSetup() throws Exception {
+		NarayanaJtaTransactionManager tm = getNarayanaJtaTransactionManager();
+		tm.afterPropertiesSet();
+		TransactionManagerImple ntm = (TransactionManagerImple)tm.getTransactionManager();
+		assertNotNull(ntm); // assert that transaction manager is a Narayana TransactionManager
+
+		String narayanaServerId = arjPropertyManager.getCoreEnvironmentBean().getNodeIdentifier();
+		String tmUid = tm.getUid();
+
+
+		assertEquals(tmUid, narayanaServerId);
+		assertStatus("ACTIVE", tmUid);
+	}
+
+	@Test
+	public void testPresetTmUid() throws Exception {
+		String presetTmUid = "fakeTmUid";
+		write(TMUID_FILE, presetTmUid);
+		NarayanaJtaTransactionManager tm = getNarayanaJtaTransactionManager();
+		tm.afterPropertiesSet();
+		TransactionManager ntm = tm.getTransactionManager();
+		assertNotNull(ntm); // assert that transaction manager is a Narayana-transaction manager
+
+		String narayanaServerId = arjPropertyManager.getCoreEnvironmentBean().getNodeIdentifier();
+		assertEquals(presetTmUid, narayanaServerId);
+		assertEquals(presetTmUid, tm.getUid());
+
+		assertStatus("ACTIVE", presetTmUid);
+	}
+
+	@Test
+	public void testCleanShutdown() throws Exception {
+		NarayanaJtaTransactionManager tm = getNarayanaJtaTransactionManager();
+		tm.afterPropertiesSet();
+		TransactionManager ntm = tm.getTransactionManager();
+		assertNotNull(ntm); // assert that transaction manager is a BitronixTransactionManager
+		String tmUid = tm.getUid();
+		assertNotNull(tmUid);
+
+		assertStatus("ACTIVE", tmUid);
+
+		tm.destroy();
+		delegateTransactionManager=null;
+
+		assertStatus("COMPLETED", tmUid);
+	}
+
+	@Ignore("For a proper test, we need to set up a transaction that cannot be completed or rolled back")
```

Comment:
Het zou gaaf zijn als btm en narayana de zelfde baseclass kunnen gebruiken.

Ik zie hier, en bij BTM een ignore. Testen we dit wel goed want hier gaat het natuurlijk eigenlijk om..